src/Drupal/DKANExtension/Context/WorkflowContext.php

```diff
@@ -30,23 +35,27 @@ public function transitionModerationState($named_entity, $state, $date = null) {
       throw new \Exception("State '$state' is not available. All possible states are [$possible_states].");
     }
 
-    $current_user = $this->getCurrentUser();
+    $current_user = $user = ($workflow_user) ? user_load_by_name($workflow_user) : $this->getCurrentUser();
     if (!$current_user) {
       throw new \Exception("No user is logged in.");
     }
 
-    $my_revision = $node->workbench_moderation['my_revision'];
-    $next_states = workbench_moderation_states_next($my_revision->state, $current_user, $node);
-    if (empty($next_states)) {
-      $next_states = array();
-    }
-    if (!isset($next_states[$state_key])) {
-      $next_states = implode(", ", $next_states);
-      throw new \Exception("State '$possible_states[$state_key]' is not available to transition to. Transitions available to user '$current_user->name' are [$next_states]");
-    }
+    $my_revision = $node->workbench_moderation['my_revision'];;
+    $flipped_posible_states = array_flip($possible_states);
 
-    // This function actually updates the transition.
-    workbench_moderation_moderate($node, $state_key, $current_user->uid);
+    if($my_revision->state != $flipped_posible_states[$state]) {
```

Comment:
Some tests needs only to change the moderation time. For example: tests checking the stale draft tab. All the nodes listed there should be in the draft state but with transitioned 72 hs before the tests. Then if you are moderating to the same state but with different time, then the moderation isn't performed but the time is changed. This allow us to write the test in a consistent way.
Another approach could be create an extra step to just change the time in. Or just add the definition on top of this function. What do you think?